tsl/src/bgw_policy/reorder_api.c

```diff
@@ -125,35 +126,30 @@ policy_reorder_add(PG_FUNCTION_ARGS)
 	NameData proc_name, proc_schema, owner;
 	int32 job_id;
 	Dimension *dim;
-
-	PreventCommandIfReadOnly("add_reorder_policy()");
-
 	Interval schedule_interval = DEFAULT_SCHEDULE_INTERVAL;
 	Oid ht_oid = PG_GETARG_OID(0);
 	Name index_name = PG_GETARG_NAME(1);
 	bool if_not_exists = PG_GETARG_BOOL(2);
-	int32 hypertable_id = ts_hypertable_relid_to_id(ht_oid);
-	Hypertable *ht = ts_hypertable_get_by_id(hypertable_id);
+	Cache *hcache;
+	Hypertable *ht;
+	int32 hypertable_id;
 	Oid partitioning_type;
 	Oid owner_id;
 	List *jobs;
 
-	owner_id = ts_hypertable_permissions_check(ht_oid, GetUserId());
+	PreventCommandIfReadOnly("add_reorder_policy()");
```

Comment:
```suggestion
	PreventCommandIfReadOnly(get_func_name(FC_FN_OID(fcinfo)));
```

Reply:
This looks like a good idea, but let's do as a separate refactoring since the situation is similar in many places and doesn't change the current errors.